Makefile

```diff
@@ -213,41 +213,35 @@ pcrs: workflows/providers/* workflows/collections/* workflows/rules/*
 	scripts/deploy-pcrs.sh ${SELF_DIR}/workflows ${cumulus_id_rsa-"~/.ssh/id_rsa"}
 
 # ------ Cumulus Dashboard ------
-#
-#  Deploying the dashboard requires the environment variables:
-#
-#    CUMULUS_API_ROOT: The HTTP URL for the Cumulus instance's API, e.g.:
-#
-#      CUMULUS_API_ROOT="https://mlcjs8s9ac.execute-api.us-east-1.amazonaws.com/dev/"
-#
-#    CUMULUS_DASHBOARD_VERSION: The version number (e.g., v1.7.2) to build & deploy
-#
-#    Then call:
-#
-#      make dashboard
-#
-tmp:
-	mkdir -p tmp
-
-tmp-cumulus-dashboard: tmp
-	rm -rf tmp/cumulus-dashboard
-	git clone https://github.com/nasa/cumulus-dashboard tmp/cumulus-dashboard
-	cd tmp/cumulus-dashboard
-	git fetch origin ${CUMULUS_DASHBOARD_VERSION}:refs/tags/${CUMULUS_DASHBOARD_VERSION}
-	git checkout ${CUMULUS_DASHBOARD_VERSION}
-
-build-dashboard: tmp-cumulus-dashboard
-	cd tmp/cumulus-dashboard
-	SERVED_BY_CUMULUS_API=${SERVED_BY_CUMULUS_API} \
-	DAAC_NAME=${DEPLOY_NAME} \
-	STAGE=${MATURITY} \
-	HIDE_PDR=false \
-	LABELS=daac \
-	APIROOT=${CUMULUS_API_ROOT} \
-	./bin/build_in_docker.sh
-
-deploy-dashboard: dashboard-init
-	cd tmp/cumulus-dashboard
-	aws s3 sync dist s3://${DEPLOY_NAME}-cumulus-${MATURITY}-dashboard --acl public-read
-
-dashboard: build-dashboard deploy-dashboard
+# Environment variables:
+#  - DASHBOARD_DIR: Directory where the dashboard source code is checked out
+
+# We could get more granular with the dependencies here, but using the
+# dashboard directory is probably fine since we aren't developing it.
+${DASHBOARD_DIR}/dist: cumulus-dashboard cumulus-init
+	if [ "${MATURITY}" = "dev" ]; then
+		export SERVED_BY_CUMULUS_API=true
+	fi
+	export DAAC_NAME=${DEPLOY_NAME}
+	export STAGE=${MATURITY}
+	export HIDE_PDR=false
+	export LABELS=daac
+	export AUTH_METHOD=launchpad
+	if [ -z "${APIROOT}" ]; then
+		export APIROOT=$(shell cd cumulus && terraform output archive_api_uri)
+	fi
+	cd $(DASHBOARD_DIR)
+	@echo "SERVED_BY_CUMULUS_API='$$SERVED_BY_CUMULUS_API'"
+	@echo "DAAC_NAME='$$DAAC_NAME'"
+	@echo "STAGE='$$STAGE'"
+	@echo "HIDE_PDR='$$HIDE_PDR'"
+	@echo "LABELS='$$LABELS'"
+	@echo "AUTH_METHOD='$$AUTH_METHOD'"
+	@echo "APIROOT='$$APIROOT'"
+	npm install --no-optional --cache ../.npm
+	npm run build
+
+.PHONY: dashboard
+dashboard: $(DASHBOARD_DIR)/dist
+	$(banner)
+	aws s3 sync $(DASHBOARD_DIR)/dist s3://${DEPLOY_NAME}-cumulus-${MATURITY}-dashboard --delete
```

Comment:
What is the `--delete` for?

Reply:
Because whenever you recompile, the file names of a bunch of the resources get set based on their md5 sums, so the `--delete` just makes it so we delete any old resources from the previous version of the dashboard. Mostly just to keep the bucket size down and so you don't get flooded with hundreds of unused .js files when you list the bucket.